src/components/Identity/configurePayload.js

```diff
@@ -0,0 +1,39 @@
+import getBrowser from "../../utils/getBrowser";
+import areThirdPartyCookiesSupportedByDefault from "../../utils/areThirdPartyCookiesSupportedByDefault";
+import ecidNamespace from "../../constants/ecidNamespace";
+
+const addEcidToPayload = (payload, ecid) => {
+  payload.addIdentity(ecidNamespace, {
+    id: ecid
+  });
+};
+export default (
```

Comment:
It would be so cool to convert this module to a factory, since `migration, idMigrationEnabled & thirdPartyCookiesEnabled` are constants. The only thing that change is the `payload`. So a function taking the first 3 args that returns a function that takes the `payload`, and you can pass this second function around instead of passing those 4 args like on line 64 & 146 in `createComponent.js`